python/lsst/cbp/coordinateConverterConfig.py

```diff
@@ -0,0 +1,217 @@
+        :ref:`pupil plane <lsst.cbp.pupil_position>`
+        from the center of the telescope
+        (mm, positive if closer to the sky)
+    telPupilDiameter : `float`
+        Diameter of telescope pupil (mm)
+    telPupilObscurationDiameter : `float`
+        Diameter of the telescope central obscuration at the
+        `pupil plane <lsst.cbp.pupil_position>` (mm)
+    telFocalPlaneDiameter : `float`
+        Diameter of telescope flocal plane (mm)
+    telFlipX : `bool`
+        True if the x axis of the telescope focal plane
+        is :ref:`flipped <lsst.cbp.flipped_x_axis>`
+        with respect to the pupil frame
+    telAzimuthOffset : `float`
+        Azimuth `offset`_ (degrees); defaults to 0
+    telAzimuthScale : `float`
+        Azimuth `scale`_; defaults to 1; must be ±1
+        (in order to handle wrap correctly).
+    telAltitudeOffset : `float` (optional)
+        Telescope altitude `offset`_ (degrees); defaults to 0
+    telAltitudeScale : `float` (optional)
+        Telescope altitude `scale`_; defaults to 1
+    telAltitudeLimits : pair of `float`
+        Telescope minimum, maximum allowed observed altitude (degrees)
+    telRotOffset : `float` (optional)
+        Telescope camera rotator offset (degrees); defaults to 0
+    telRotScale : `float`
+        Telescope camera rotator scale; defaults to 1; must be ±1
+        (in order to handle wrap correctly)
+    defaultDetector : `str`
+        Name of default detector
+
+    cbpPosition : triplet of `float`
+        CBP x, y, z position of center of CBP relative to the center
+        of the telescope, in the base frame (mm)
+    cbpFocalLength : `float`
+        Effective focal length of the CBP (mm);
+        635 mm is an estimate for the LSST's CBP
+    cbpFlipX : `bool`
+        True if the x axis of the CBP focal plane is flipped
+        with respect to the pupil frame?
+    cbpAzimuthOffset : `float` (optional)
+        CBP azimuth `offset`_ (degrees); defaults to 0
+    cbpAzimuthScale : `float` (optional)
+        CBP azimuth `scale`_; defaults to 1; must be ±1
+        (in order to handle wrap correctly)
+    cbpAltitudeOffset : `float` (optional)
+        CBP altitude `offset`_ (degrees); defaults to 0
+    cbpAltitudeScale : `float` (optional)
+        CBP altitude `scale`_; defaults to 1
+    cbpAltitudeLimits : pair of `float`
+        CBP minimum, maximum allowed observed altitude (degrees)
+
+    Raises
+    ------
+    AssertionError
+        Raised if ``telAzimuthScale``, ``cbpAzimuthScale``
+        and/or ``telRotScale`` is not ±1
+
+    Notes
+    -----
+
+    .. _offset:
+
+    .. _scale:
+
+    **Offset and Scale:**
+
+    Azimuth, altitude and rotator offset and scale define the mapping
+    between :ref:`internal angle <lsst.cbp.internal_angles>`
+    and :ref:`observed angle <lsst.cbp.observed_angles>` as follows::
+
+        observed angle = internal angle * scale + offset
+
+    .. _fields:
+
+    **Fields:**
+
+    telPupilOffset : `float`
```

Comment:
I agree it's a nuisance. From reading the generated docs it does appear to be necessary. There are a few subtle differences between inputs and fields, but most of the information is just repetition, which is verbose and difficult to maintain.

Reply:
Personally I always favour longer and more thorough argument names, length be damned, and if we're mixing deg and rad I think that might be quite nice actually, but only if you can be bothered to make the changes.

Reply:
I agree. I have appended "Deg" to the altitude, azimuth and rotator offset arguments and the altitude limits arguments and I think it does improve clarity. I also found and fixed a bug: I was using Angles instead of floats as the defaults.